Projects/Server/MultiGameRunner.cpp

```diff
@@ -0,0 +1,60 @@
+#include "MultiGameRunner.h"
+
+MultiGameRunner::MultiGameRunner(const AbstractPlayer::Ptr& player0, const AbstractPlayer::Ptr& player1, const size_t& totalGamesToBePlayed = 1)
+	: _player0{ player0 }, _player1{ player1 }, _totalGamesToBePlayed{ totalGamesToBePlayed }
+{
+	_gameOutcomes[State::Outcome::P0WIN];
+	_gameOutcomes[State::Outcome::P1WIN];
+	_gameOutcomes[State::Outcome::DRAW];
+	_currentGameCount = 0;
+}
+
+
+void MultiGameRunner::play() 
+{
+	GameHandler gameHandler{};
+
+	for (size_t i{ 0 }; i < _totalGamesToBePlayed; ++i) 
+	{	
+		gameHandler.init(_player0, _player1);
+		
+
+		State::Outcome outcome = gameHandler.play();
+		
+		++_currentGameCount;
```

Comment:
What is the point of having `_currentGameCount` as a member field, if it can never be other than 0 or `_totalGamesToBePlayed`?
Related to that, there is no point having the method mutable, if you can't inspect the games anyway. If you just want to have two players play against each other N times, and get win percentages, then you should have this `play()` method return the `_gameOutcomes`, and only the players + `_totalGamesToBePlayed` should be members.

Reply:
What I had in mind was that,it would print the current state at the end of each game through `operator<<`. Thus, it could be anything between 0 or `_totalGamesToBePlayed`. However, I realized that, I forgot to add it inside the `GameRunner::play()` and, it just prints at `main` only because it's a leftover from my tests.

It is now fixed to reflect what I intended.

Reply:
Still, even now there is no reason to track `_currentGameCount`. It will be just equal to `i`.

Reply:
It is removed and implemented a new function which announces the results at the end of `MultiGameRunner::play()`.